### ISSUE TITLE
Does yolov5 use multiscale training ?

### ISSUE THREAD

**Issue by UygarUsta99** [ultralytics/yolov5#6291]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

Like I have stated in the title,does yolov5 use multiscale training(at various image sizes) ? It is often stated that using multiscale training helps in detecting small objects besides training them with a fixed high resolution.Thanks for all the help.

**Comment by glenn-jocher** [comment]
@UygarUsta99 👋 Hello! Thanks for asking about **image augmentation**. `scale=0.5` hyperparameter controls scale jitter. YOLOv5 🚀 applies online imagespace and colorspace augmentations in the trainloader (but not the val_loader) to present a new and unique augmented [Mosaic](https://github.com/ultralytics/yolov5/blob/90b7895d652c3bd3d361b2d6e9aee900fd67f5f7/utils/datasets.py#L678-L732) (original image + 3 random images) each time an image is loaded for training. **Images are never presented twice in the same way**.

![YOLOv5 augmentation](https://user-images.githubusercontent.com/26833433/120995721-f3cfed00-c785-11eb-8ee2-b6ef2fa205e8.jpg)

### Augmentation Hyperparameters

The hyperparameters used to define these augmentations are in your hyperparameter file (default `data/hyp.scratch.yaml`) defined when training:
```
python train.py --hyp hyp.scratch.yaml
```
https://github.com/ultralytics/yolov5/blob/90b7895d652c3bd3d361b2d6e9aee900fd67f5f7/data/hyp.scratch.yaml#L1-L33


### Augmentation Previews

You can view the effect of your augmentation policy in your train_batch*.jpg images once training starts. These images will be in your train logging directory, typically `yolov5/runs/train/exp`:

`train_batch0.jpg` shows train batch 0 mosaics and labels:  
<img src="https://user-images.githubusercontent.com/26833433/131255960-b536647f-7c61-4f60-bbc5-cb2544d71b2a.jpg" width="800">


### YOLOv5 Albumentations Integration

[YOLOv5](https://github.com/ultralytics/yolov5) 🚀 is now fully integrated with [Albumentations](https://github.com/albumentations-team/albumentations), a popular open-source image augmentation package. Now you can train the world's best Vision AI models even better with custom Albumentations 😃! 

PR https://github.com/ultralytics/yolov5/pull/3882 implements this integration, which will automatically apply Albumentations transforms during YOLOv5 training if `albumentations>=1.0.3` is installed in your environment. See https://github.com/ultralytics/yolov5/pull/3882 for full details.

Example `train_batch0.jpg` on COCO128 dataset with Blur, MedianBlur and ToGray. See the YOLOv5 [Notebooks](https://github.com/ultralytics/yolov5/blob/master/tutorial.ipynb) to reproduce: <a href="https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolov5"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>

<img src="https://user-images.githubusercontent.com/26833433/124400879-ff331b80-dd25-11eb-9b67-fe85ac4ca104.jpg" width="800">

Good luck 🍀 and let us know if you have any other questions!

**Comment by ankur219** [comment]
@glenn-jocher, I think he meant multi-scale training where you change the image size and not the scale augmentation. Just wanted to confirm. Thanks!

**Comment by glenn-jocher** [comment]
@ankur219 you can change image size with the --multi-scale arg (not recommended).
```
python train.py --multi-scale
```

https://github.com/ultralytics/yolov5/blob/0b5ac224aef287ac3ac9ebf70ade60159450a0b1/train.py#L468

**Comment by alxgrod** [comment]
> @ankur219 you can change image size with the --multi-scale arg (not recommended).
> 
> ```
> python train.py --multi-scale
> ```
> 
> https://github.com/ultralytics/yolov5/blob/0b5ac224aef287ac3ac9ebf70ade60159450a0b1/train.py#L468

Why is not recommended?

**Comment by glenn-jocher** [comment]
worse results

**Comment by ch-hristov** [comment]
@glenn-jocher Citing from a paper I've been reading: "Adjusting the input size during training was reported to improve object detection across different object scales [0]". 

I might be missing something here but AFAIK the darknet implementation had an argument --random which did multi scale training and the paper above reports that the results are improved this way. Did you mean that in practice you found this to be not true?

[0]: YOLO9000: Better, Faster, Stronger - https://arxiv.org/pdf/1612.08242.pdf

**Comment by glenn-jocher** [comment]
@ch-hristov YOLOv5 by default uses 50% scale jitter, which changes the scale of the image contents, you can modify with the scale hyp. --multi-scale will also change the actual image size, default is False.

**Comment by pramodsarvi** [comment]
@glenn-jocher How does YOLOv5 handle multiple image sizes? Can you please guide me how the network is independent of image shape.
